Assets/MRTK/Examples/Experimental/ColorPicker/Scripts/ColorPicker.cs

```diff
@@ -0,0 +1,352 @@
+        [Tooltip("")]
+        private PinchSlider SliderGreen = null;
+        [SerializeField]
+        [Tooltip("")]
+        private PinchSlider SliderBlue = null;
+        [SerializeField]
+        [Tooltip("")]
+        private PinchSlider SliderAlpha = null;
+        [SerializeField]
+        [Tooltip("")]
+        private PinchSlider SliderHue = null;
+        [SerializeField]
+        [Tooltip("")]
+        private PinchSlider SliderSaturation = null;
+        [SerializeField]
+        [Tooltip("")]
+        private PinchSlider SliderBrightness = null;
+        //
+        [SerializeField]
+        [Tooltip("")]
+        private TextMeshPro TextRed = null;
+        [SerializeField]
+        [Tooltip("")]
+        private TextMeshPro TextGreen = null;
+        [SerializeField]
+        [Tooltip("")]
+        private TextMeshPro TextBlue = null;
+        [SerializeField]
+        [Tooltip("")]
+        private TextMeshPro TextAlpha = null;
+        [SerializeField]
+        [Tooltip("")]
+        private TextMeshPro TextHex = null;
+        [SerializeField]
+        [Tooltip("")]
+        private TextMeshPro TextHue = null;
+        [SerializeField]
+        [Tooltip("")]
+        private TextMeshPro TextSaturation = null;
+        [SerializeField]
+        [Tooltip("")]
+        private TextMeshPro TextBrightness = null;
+        //
+        private float GradientDragMaxDistance = 0.5f;
+        private Vector3 GradientDragStartPosition;
+        private Vector3 GradientDragCurrentPosition;
+        //
+        private Color CustomColor;
+        private float Hue, Saturation, Brightness, Alpha = 0.3f;
+        //
+        private bool IsDraggingSliders = false;
+        private bool IsDraggingGradient = false;
+        //
+        private void Start()
+        {
+            GradientDragStartPosition = GradientDragger.transform.localPosition;
+            GradientDragCurrentPosition = GradientDragStartPosition;
+            this.gameObject.SetActive(false);
+        }
+        private void Update()
+        {
+            if (IsDraggingGradient)
+            {
+                ConstrainGradientDragging();
+            }
+            if (IsDraggingSliders)
+            {
+                CalculateGradientDraggerPosition();
+            }
+        }
+        #region Gradient Logic (Private)
+        void IMixedRealityTouchHandler.OnTouchUpdated(HandTrackingInputEventData eventData)
+        {
+            //OnTouchUpdated.Invoke(eventData);
+            //Debug.Log("OnTouchUpdated: " + Time.unscaledTime);
+            GradientDragger.transform.position = new Vector3(eventData.InputData.x, eventData.InputData.y, eventData.InputData.z);
+            ConstrainGradientDragging();
+            ApplyColor();
+            UpdateSliderText();
+            ApplySliderValues();
+        }
+        void IMixedRealityTouchHandler.OnTouchStarted(HandTrackingInputEventData eventData)
+        {
+            //OnTouchStarted.Invoke(eventData);
```

Comment:
Do you want to keep this commented code around for reference later, or would you be okay to remove the comments?

Reply:
removed